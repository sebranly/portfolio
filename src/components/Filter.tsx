import * as React from 'react';
import { Tag as TagType } from '../types';
import { getAllTags, getAllYears } from '../utils';
import { Tag } from './Tag';
import { projects } from '../data/projects';

export interface FilterProps {
  className?: string;
}

const Filter: React.FC<FilterProps> = (props) => {
  const { className } = props;
  const allYears = getAllYears(projects);
  const tags = getAllTags(projects);
  const allTags = [...allYears, ...tags];

  return (
    <div className={className}>
      {allTags.map((tag: TagType | number) => {
        return <Tag key={tag} tag={tag} />;
      })}
    </div>
  );
};

export { Filter };
