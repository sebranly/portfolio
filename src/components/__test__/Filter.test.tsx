import * as React from 'react';
import { render } from '@testing-library/react';

import { Filter } from '../Filter';

test('Filter', () => {
  const createComponent = () => <Filter />;

  const { container } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot();
});
