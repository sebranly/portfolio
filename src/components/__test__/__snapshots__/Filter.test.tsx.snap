// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter 1`] = `
NodeList [
  <div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-black hover:bg-gray-700"
    >
      2011
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-black hover:bg-gray-700"
    >
      2012
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-black hover:bg-gray-700"
    >
      2013
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-black hover:bg-gray-700"
    >
      2014
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-red-500 hover:bg-red-600"
    >
      3D
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-gray-500 hover:bg-gray-600"
    >
      projects.general.tags.archived
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-yellow-500 hover:bg-yellow-600"
    >
      C
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-indigo-500 hover:bg-indigo-600"
    >
      CSS
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-orange-500 hover:bg-orange-600"
    >
      HTML
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-yellow-500 hover:bg-yellow-600"
    >
      JavaScript
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-green-500 hover:bg-green-600"
    >
      myProjects
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-red-500 hover:bg-red-600"
    >
      Nintendo DS
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-indigo-500 hover:bg-indigo-600"
    >
      OpenGL
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-orange-500 hover:bg-orange-600"
    >
      PAlib
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-indigo-500 hover:bg-indigo-600"
    >
      PHP
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-blue-500 hover:bg-blue-600"
    >
      SDL
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-orange-500 hover:bg-orange-600"
    >
      SQL
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-amber-500 hover:bg-amber-600"
    >
      projects.general.tags.video game
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-purple-500 hover:bg-purple-600"
    >
      projects.general.tags.website
    </div>
    <div
      class="border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-blue-500 hover:bg-blue-600"
    >
      Windows
    </div>
  </div>,
]
`;
